portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,80 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
+import com.google.gson.Gson;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import java.text.SimpleDateFormat;  
+import java.util.Date; 
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Ryan!");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    String numComments = request.getParameter("num-comments");
+    int commentAmount = 0;
+    try{
+        commentAmount = Integer.parseInt(numComments);
+    }catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + numComments);
+      commentAmount = 5;
+    }
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(commentAmount))) {
+      String content = (String) entity.getProperty("content");
+      String date = (String) entity.getProperty("date");
+      long timestamp = (long) entity.getProperty("timestamp");
+      long id = entity.getKey().getId();
+
+      Comment comment = new Comment(content, timestamp, id, date);
+      comments.add(comment);
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
```

Comment:
```gson.toJson(comments, gson.newJsonWriter(response.getWriter()));```

Reply:
This produces a compilation error
![image](https://user-images.githubusercontent.com/51490436/84039636-89684f00-a96f-11ea-9712-7ff12228dd8f.png)